### PR TITLE
sl-67: Stepper  step idustry

### PR DESCRIPTION
**Pull request by nikita-kulchytskyi**
image

### CODE REVIEW

File: frontend/src/pages/start/components/step-industry/step-industry.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import React, { useCallback } from 'react';
```
Comment (jaroslawtt) [inline frontend/src/pages/start/components/step-industry/step-industry.tsx]:
```suggestion

```

File: frontend/src/pages/start/components/step-industry/step-industry.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+import { Input, Select } from '~/libs/components/components.js';
+import { useAppForm } from '~/libs/hooks/hooks.js';
```
Comment (jaroslawtt) [inline frontend/src/pages/start/components/step-industry/step-industry.tsx]:
```suggestion
import { useAppForm, useCallback } from '~/libs/hooks/hooks.js';
```

File: frontend/src/pages/start/start.tsx [hunk 3]
```diff
@@ -1,8 +1,16 @@
+import { useCallback } from 'react';
```
Comment (jaroslawtt) [inline frontend/src/pages/start/start.tsx]: same

File: frontend/src/pages/start/components/step-industry/step-industry.tsx (hunk 2 above)
Comment (nikita-kulchytskyi) [reply frontend/src/pages/start/components/step-industry/step-industry.tsx]: Done!

File: frontend/src/libs/components/stepper/stepper.tsx [hunk 4]
```diff
@@ -72,14 +72,12 @@ const Stepper: React.FC<Properties> = ({
...
       <Button
+        type="submit"
         label={buttonLabel}
         style="secondary"
         size="small"
         onClick={handleNextStep}
-        className={getValidClassNames(
-          styles['button'],
-          isLastStep && 'visually-hidden',
-        )}
```
Comment (Alyona-Pisotska) [inline frontend/src/libs/components/stepper/stepper.tsx]: You don`t need to delete this logic. In the last step "save button" will be the part of last step component

File: frontend/src/pages/start/components/step-industry/step-industry.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        <Select
+          control={control}
+          errors={errors}
+          name="industrySelect"
+          placeholder="Please select your industry"
+          options={options}
+        />
+        <Input
+          type="text"
+          label=""
```
Comment (Alyona-Pisotska) [inline frontend/src/pages/start/components/step-industry/step-industry.tsx]: A11

**Review by Alyona-Pisotska — CHANGES_REQUESTED** [review]

File: frontend/src/libs/components/stepper/stepper.tsx (hunk 4 above)
Comment (nikita-kulchytskyi) [reply frontend/src/libs/components/stepper/stepper.tsx]: It doesn't work correctly until I removed this part,

File: frontend/src/pages/start/components/step-industry/step-industry.tsx (hunk 5 above)
Comment (nikita-kulchytskyi) [reply frontend/src/pages/start/components/step-industry/step-industry.tsx]: Done!

File: frontend/src/libs/components/stepper/stepper.tsx (hunk 4 above)
Comment (nikita-kulchytskyi) [reply frontend/src/libs/components/stepper/stepper.tsx]: Returned back and also returned the bar

File: frontend/src/libs/components/button/button.tsx [hunk 6]
```diff
@@ -7,6 +7,7 @@ import styles from './styles.module.scss';
 
 type Properties = {
   label: string;
+  form?: string;
```
Comment (what1s1ove) [inline frontend/src/libs/components/button/button.tsx]:
```suggestion
```

File: frontend/src/libs/components/stepper/stepper.tsx [hunk 7]
```diff
@@ -72,6 +72,8 @@ const Stepper: React.FC<Properties> = ({
       </div>
 
       <Button
+        form="test-step"
```
Comment (what1s1ove) [inline frontend/src/libs/components/stepper/stepper.tsx]:
```suggestion
```

File: frontend/src/pages/start/components/step-industry/libs/constants.tsx [hunk 8]
```diff
@@ -0,0 +1,8 @@
+import { type StepIndustryDto } from '~/packages/stepper/validation-schemas/validation-schema.js';
+
+const DEFAULT_STEP_INDUSTRY_VALUE: StepIndustryDto = {
+  industrySelect: { value: '', label: '' },
+  industryInput: '',
```
Comment (what1s1ove) [inline frontend/src/pages/start/components/step-industry/libs/constants.tsx]: give a better name for it. both should be a string

File: shared/src/packages/stepper/libs/validation-schemas/step-industry.validation-schema.ts [hunk 9]
```diff
@@ -0,0 +1,32 @@
+import joi from 'joi';
```
Comment (what1s1ove) [inline shared/src/packages/stepper/libs/validation-schemas/step-industry.validation-schema.ts]: it should be a part of `site` packag

File: shared/src/packages/stepper/libs/types/step-industry-dto.type.ts [hunk 10]
```diff
@@ -0,0 +1,11 @@
+type StepIndustryDto = {
```
Comment (what1s1ove) [inline shared/src/packages/stepper/libs/types/step-industry-dto.type.ts]: each type should be in a separate file

File: frontend/src/pages/start/start.tsx [hunk 11]
```diff
@@ -1,8 +1,21 @@
+import { type StepIndustryDto } from 'shared/build/index.js';
```
Comment (what1s1ove) [inline frontend/src/pages/start/start.tsx]: change import

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/libs/components/button/button.tsx (hunk 6 above)
Comment (nikita-kulchytskyi) [reply frontend/src/libs/components/button/button.tsx]: Without this, I don't know how to trigger the validation on the next button

File: frontend/src/pages/start/start.tsx (hunk 11 above)
Comment (nikita-kulchytskyi) [reply frontend/src/pages/start/start.tsx]: Done

File: frontend/src/pages/start/start.tsx [hunk 12]
```diff
@@ -1,8 +1,21 @@
...
+
 import { PageLayout, Stepper } from '~/libs/components/components.js';
+import { useCallback } from '~/libs/hooks/hooks.js';
 
+import { StepIndustry } from './components/step-industry/step-industry.js';
 import styles from './styles.module.scss';
 
 const Start: React.FC = () => {
+  const handlesStepSubmit = useCallback((payload: StepIndustryDto): void => {
```
Comment (bohdan-degtiar) [inline frontend/src/pages/start/start.tsx]:
```suggestion
  const handleStepSubmit = useCallback((payload: StepIndustryDto): void => {
```

File: frontend/src/pages/start/start.tsx [hunk 13]
```diff
@@ -1,8 +1,21 @@
...
+    payload;
+    //idle function, just to push
+  }, []);
+  const steps = [
+    <StepIndustry key="secondStep" onSubmit={handlesStepSubmit} />,
+    <StepIndustry key="secondStep" onSubmit={handlesStepSubmit} />,
```
Comment (bohdan-degtiar) [inline frontend/src/pages/start/start.tsx]: StepIndustry twice?

File: shared/src/packages/stepper/libs/enums/enums.ts [hunk 14]
```diff
@@ -0,0 +1,2 @@
+export { StepperValidationMessage } from './stepper-validation-message.enum.js';
+export { StepIndustryValidationRule } from './stepper-validation-rule.enum.js';
```
Comment (bohdan-degtiar) [inline shared/src/packages/stepper/libs/enums/enums.ts]: StepIndustryValidationRule but file name stepper-validation-rule.enum.js

File: frontend/src/pages/start/components/step-industry/libs/constants.tsx [hunk 15]
```diff
@@ -0,0 +1,8 @@
...
+};
```
Comment (bohdan-degtiar) [inline frontend/src/pages/start/components/step-industry/libs/constants.tsx]:
```suggestion
} as const;
```

File: shared/src/packages/stepper/libs/validation-schemas/step-industry.validation-schema.ts [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+
+import {
+  StepIndustryValidationRule,
+  StepperValidationMessage,
```
Comment (bohdan-degtiar) [inline shared/src/packages/stepper/libs/validation-schemas/step-industry.validation-schema.ts]: Unify this names.

File: frontend/src/pages/start/start.tsx [hunk 17]
```diff
@@ -1,8 +1,21 @@
...
+import { StepIndustry } from './components/step-industry/step-industry.js';
 import styles from './styles.module.scss';
 
 const Start: React.FC = () => {
+  const handlesStepSubmit = useCallback((payload: StepIndustryDto): void => {
+    payload;
+    //idle function, just to push
+  }, []);
+  const steps = [
+    <StepIndustry key="secondStep" onSubmit={handlesStepSubmit} />,
```
Comment (bohdan-degtiar) [inline frontend/src/pages/start/start.tsx]:
```suggestion
    <StepIndustry key="secondStep" onStepSubmit={handlesStepSubmit} />,
```

**Review by bohdan-degtiar — CHANGES_REQUESTED** [review]

File: frontend/src/pages/start/libs/components/industry-form/industry-form.tsx [hunk 18]
```diff
@@ -0,0 +1,86 @@
...
+import { type SiteCreateIndustryName } from '~/packages/sites/sites.js';
+import { siteCreateStepIndustryValidationSchema } from '~/packages/sites/sites.js';
+
+import { DEFAULT_STEP_PAYLOAD } from './libs/constants.js';
+import styles from './styles.module.scss';
+
+type Properties = {
+  onSubmit: (sitePayload: Partial<SiteCreateIndustryName>) => void;
+};
+const IndustryForm: React.FC<Properties> = ({ onSubmit }: Properties) => {
```
Comment (bohdan-degtiar) [inline frontend/src/pages/start/libs/components/industry-form/industry-form.tsx]:
```suggestion
};

const IndustryForm: React.FC<Properties> = ({ onSubmit }: Properties) => {
```

File: shared/src/packages/sites/libs/validation-schemas/site-create-step-industry.validation-schema.ts [hunk 19]
```diff
@@ -0,0 +1,22 @@
...
+    }),
+    enterIndustry: joi
+      .string()
+      .pattern(SiteValidationRule.SITE_INDUSTRY_REGEX)
+      .messages({
+        'string.empty': SiteValidationMessage.SITE_INDUSTRY_REQUIRE,
+        'string.pattern.base': SiteValidationMessage.SITE_INDUSTRY_WRONG,
+      }),
+  })
+  .or('selectIndustry', 'enterIndustry');
```
Comment (bohdan-degtiar) [inline shared/src/packages/sites/libs/validation-schemas/site-create-step-industry.validation-schema.ts]: use `xor` and add empty value for `selectIndustry` and `enterIndustry`

**Review by bohdan-degtiar — CHANGES_REQUESTED** [review]

File: shared/src/packages/sites/libs/validation-schemas/site-create-step-industry.validation-schema.ts (hunk 19 above)
Comment (nikita-kulchytskyi) [reply shared/src/packages/sites/libs/validation-schemas/site-create-step-industry.validation-schema.ts]: @bohdan-degtiar its working but without errors messages (empty messages, pattern working) Errors are not included in the useform, when use empty(' ')
image

File: frontend/src/libs/components/button/button.tsx (hunk 6 above)
Comment (andanf-e) [reply frontend/src/libs/components/button/button.tsx]: But you haven't passed this prop anywhere 🤔

File: frontend/src/pages/start/libs/components/industry-form/industry-form.tsx [hunk 20]
```diff
@@ -0,0 +1,87 @@
...
+  const [disableSelect, setDisableSelect] = useState(false);
+  const [disableInput, setDisableInput] = useState(false);
+
+  useEffect(() => {
+    return () => {
+      const { selectIndustry, enterIndustry } = watchValue;
+      setDisableSelect(Boolean(enterIndustry));
+      setDisableInput(Boolean(selectIndustry));
+    };
+  }, [watchValue]);
```
Comment (andanf-e) [inline frontend/src/pages/start/libs/components/industry-form/industry-form.tsx]: It would disable fields after this component would be unmounted?

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**